src/PackageVersions/Installer.php

```diff
@@ -137,8 +139,10 @@ private static function writeVersionClassToFile(string $versionClassSource, Comp
 
         $io->write('<info>ocramius/package-versions:</info>  Generating version class...');
 
-        file_put_contents($installPath, $versionClassSource);
-        chmod($installPath, 0664);
+        $installPathTmp = $installPath . '_' . uniqid('tmp', true);
```

Comment:
if you want really to be secure, you should not be using [uniqid](https://php.net/uniqid) but [tempnam](https://php.net/tempnam) which is aware of filesystem race conditions.

Reply:
Please open a new issue about that, if needed to be fixed. I don't expect a concurrency > 1 for `PackageVersions` specifically